server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessAttributeValueHealth.java

```diff
@@ -0,0 +1,336 @@
+    }
+
+    public int getActiveShards() {
+        return activeShards;
+    }
+
+    public void setActiveShards(int activeShards) {
+        this.activeShards = activeShards;
+    }
+
+    public int getUnassignedShards() {
+        return unassignedShards;
+    }
+
+    public void setUnassignedShards(int unassignedShards) {
+        this.unassignedShards = unassignedShards;
+    }
+
+    public int getNodes() {
+        return nodes;
+    }
+
+    public void setNodes(int nodes) {
+        this.nodes = nodes;
+    }
+
+    public double getWeight() {
+        return weight;
+    }
+
+    public void setWeight(double weight) {
+        this.weight = weight;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getInitializingShards() {
+        return initializingShards;
+    }
+
+    public void setInitializingShards(int initializingShards) {
+        this.initializingShards = initializingShards;
+    }
+
+    public int getRelocatingShards() {
+        return relocatingShards;
+    }
+
+    public void setRelocatingShards(int relocatingShards) {
+        this.relocatingShards = relocatingShards;
+    }
+
+    public List<String> getNodeList() {
+        return nodeList;
+    }
+
+    public void setNodeList(List<String> nodeList) {
+        this.nodeList = nodeList;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(name);
+        out.writeVInt(activeShards);
+        out.writeVInt(initializingShards);
+        out.writeVInt(relocatingShards);
+        out.writeVInt(unassignedShards);
+        out.writeVInt(nodes);
+        out.writeDouble(weight);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(getName());
+        builder.field(ACTIVE_SHARDS, getActiveShards());
+        builder.field(INITIALIZING_SHARDS, getInitializingShards());
+        builder.field(RELOCATING_SHARDS, getRelocatingShards());
+        builder.field(UNASSIGNED_SHARDS, getUnassignedShards());
+        builder.field(NODES, getNodes());
+        builder.field(WEIGHTS, getWeight());
+        builder.endObject();
+        return builder;
+    }
+
+    void computeAttributeValueLevelInfo(ClusterState clusterState, boolean displayUnassignedShardLevelInfo, int shardsPerAttributeValue) {
+        // computing nodes info
+        nodes = nodeList.size();
+
+        // computing shards into
+        setShardLevelInfo(clusterState, displayUnassignedShardLevelInfo, shardsPerAttributeValue);
+
+        // compute weight info
+        setWeightInfo(clusterState);
+    }
+
+    private void setShardLevelInfo(ClusterState clusterState, boolean displayUnassignedShardLevelInfo, int shardsPerAttributeValue) {
+        String nodeId;
+
+        // computing active, relocating and initializing shards info
+        for (ShardRouting shard : clusterState.routingTable().allShards()) {
```

Comment:
We don't need to iterate over all shards for all distinct attribute value, its a wasted computation. Instead we could filter the `RoutingNode` for this attribute value and pass it in the the ctor and iterate over only those set of nodes. This would also avoid the check for `nodeInCurrentAZ `